crates/env/src/call/create_builder.rs

```diff
@@ -339,54 +365,97 @@ where
     }
 }
 
-impl<E, GasLimit, Args, Salt, R>
+impl<E, CodeHash, GasLimit, Endowment, Args, Salt>
+    CreateBuilder<E, CodeHash, GasLimit, Endowment, Args, Salt, Unset<ReturnType<()>>>
+where
+    E: Environment,
+{
+    /// Sets the type of the returned value upon the execution of the call.
+    ///
+    /// # Note
+    ///
+    /// Constructors are not able to return arbitrary values. Instead a successful call to a
+    /// constructor returns the address at which the contract was instantiated.
+    ///
+    /// Therefore this must always be a reference (i.e `ContractRef`) to the contract you're trying
+    /// to instantiate.
+    #[inline]
+    pub fn returns<R>(
+        self,
+    ) -> CreateBuilder<E, CodeHash, GasLimit, Endowment, Args, Salt, Set<ReturnType<R>>>
```

Comment:
Should we add the `R: FromAccountId<E>` constraint here too? I see we have it below on the final impl but it might give a more helpful compiler error if we do it here?